### DISCUSSION TITLE
Browser Extensions: Ability to "Find & Replace" reverse-aliases in current web page

### DISCUSSION THREAD

**Discussion by argarner** [simple-login/app#659] [Feature Requests]
**This is associated with another feature request I just raised, related to viewing reverse-aliases in the browser extensions https://github.com/simple-login/app/discussions/658**

Currently there is an ["experimental" feature](https://app.simplelogin.io/dashboard/setting#reverse-alias-replacement-section) for Simplelogin to automatically replace reverse aliases when forwarding emails. This doesn't work with all email clients (e.g. Gmail web app; meaning half the web), and many users would prefer simplelogin to not process their email bodies in any way (replacing reverse-aliases in email threads/chains), so the ability to "find & replace" reverse-aliases on email web clients (via the browser extension, and possibly a web/mobile app textbox) would be a good way to eliminate the flaky experimental feature, prevent email body processing being performed by simplelogin, and alert users when reverse aliases are visible in an email reply/forward.

Possible Requirements/Considerations:
- Implement as an optional permission ("Optional permissions for added functionality:")
- display extension badge when reverse aliases are detected in the current webpage
- clicking on extension displays dropdown with additional tab for viewing the current web pages reverse-aliases (and associated aliases)
- user can click to copy both alias & reverse-alias (if they wish to manually replace)
- badge disappears when all reverse aliases are replaced
- ability to set an allowlist so the extension only runs on user-defined domains

It'd be good if users could complete the same via the mobile app... Perhaps by copying and pasting the email text/html into a textbox in the simplelogin app, the app doing a client side "find & replace" of reverse-aliases, then the user copying & pasting the result back into the email app of their choice...

Thanks for the service! Big fan of your work!

**Comment by nguyenkims** [comment]
@argarner thanks for the suggestion. We've improved the feature "Reverse Alias Replacement" recently to take into account different email clients, can you please give it a try?

**Reply by argarner** [reply]
Thanks for the quick response @nguyenkims. I re-activated the reverse alias replacement yesterday or the day before, and gave it a forward test now. It is still not replacing my private email in forwarded threads (as I would expect it to). I have just emailed this example to you from my accounts primary email. Let me know if you need further info.

**Reply by nguyenkims** [reply]
Hi this feature only works well with new reverse-alias and unfortunately it's still experimental for existing reverse aliases ...
